generators/directive/index.js

```diff
@@ -33,115 +38,89 @@ module.exports = class extends Generator {
       type: 'input',
       name: 'controllerAs',
       message: 'Controller name?',
-      default: 'vm'
+      default: 'vm',
+      when: (answers) => {
+        return answers.element;
+      }
     }];
 
-    return this.prompt(prompts).then(function (props) {
-      // To access props later use this.props.someAnswer;
-      this.props = props;
-
-    }.bind(this));
+    return this.prompt(prompts)
+      .then((answers) => {
+        this.opts            = answers;
+        this.opts.lowerCase  = this.opts.name.toLowerCase();
+        this.opts.prefix     = this.config.get('prefix') ? this.config.get('prefix') : 'app';
+        this.opts.module     = caseIt.camelCase(`${this.opts.prefix}-${this.opts.name}-directive`);
+        this.opts.directive  = caseIt.camelCase(`${this.opts.prefix}-${this.opts.name}`);
+        this.opts.controller = caseIt.pascal(`${this.opts.name}-Ctrl`);
+        this.opts.urlSafe    = caseIt.paramCase(`${this.opts.prefix}-${this.opts.name}`);
+    });
```

Comment:
cleaned this up a bit and used var names that provide a little more context. moved it to the prompt queue because it makes a little more sense here and cleans up the writing queue some.